pkg/loadbalancer/target/metrics.go

```diff
@@ -77,6 +97,45 @@ func (hm *HitsMetrics) init() error {
 		Priority: nftables.ChainPriorityRef(-500),
 	})
 
+	err := conn.Flush()
+	if err != nil {
+		return err
+	}
+
+	// nft add map inet meridio-metrics fwmark-verdict { type mark : verdict\; }
+	hm.fwmarkVerdict = &nftables.Set{
+		Table:    hm.table,
+		Name:     setName,
+		IsMap:    true,
+		KeyType:  nftables.TypeMark,
+		DataType: nftables.TypeVerdict,
+	}
+	err = conn.AddSet(hm.fwmarkVerdict, nil)
+	if err != nil {
+		return err // shouldn't happen since elements is nil
+	}
+
+	// nft --debug all add rule inet meridio-metrics target-hits mark vmap @fwmark-verdict
```

Comment:
The rule will get visited for outbound packets as well, thus imho adding a check for non-zero marks first would lead to a better performance.